internal/utils/validation/entity_guid_validator.go

```diff
@@ -0,0 +1,64 @@
+package validation
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const (
+	DELIMITER = "|"
+)
+
+var (
+	invalidEntityGuidErr = errors.New("invalid entity GUID format")
+	emptyEntityTypeErr   = errors.New("entity type is required")
+	emptyDomainIdErr     = errors.New("domain ID is required")
+)
```

Comment:
I think the following would be nicer (maybe; just give it a thought) if possible

```suggestion
var EntityGUIDValidationErrorTypes = struct {
	INVALID_ENTITY_GUID_ERROR APIAccessIngestKeyErrorType
	EMPTY_ENTITY_TYPE_ERROR APIAccessIngestKeyErrorType
	EMPTY_DOMAIN_ID_ERROR AccessIngestKeyErrorType
}{
	INVALID_ENTITY_GUID_ERROR: errors.New("invalid entity GUID format")
	EMPTY_ENTITY_TYPE_ERROR: errors.New("entity type is required")
	EMPTY_DOMAIN_ID_ERROR: "errors.New("domain ID is required")
}

type EntityGUIDValidationError error
// or maybe have this as a string with error strings defined in types, and in the function that throws an error, use errors.New(EntityGUIDValidationError.INVALID_ENTITY_GUID_ERROR)

```